l10n_it_invoices_data_communication/models/account_invoice.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+
+
+from odoo import fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    comunicazione_dati_iva_escludi = fields.Boolean(
+        string='Exclude from invoices communication')
+
+    def _compute_taxes_in_company_currency(self, vals):
+        try:
+            exchange_rate = (
+                self.amount_total_signed /
+                self.amount_total_company_signed)
+        except ZeroDivisionError:
+            exchange_rate = 1
+        vals['ImponibileImporto'] = vals['ImponibileImporto'] / exchange_rate
+        vals['Imposta'] = vals['Imposta'] / exchange_rate
+
+    def _get_tax_comunicazione_dati_iva(self):
+        self.ensure_one()
+        fattura = self
+        tax_model = self.env['account.tax']
+
+        tax_lines = []
+        tax_grouped = {}
+        for tax_line in fattura.tax_line_ids:
+            tax = tax_line.tax_id
+            aliquota = tax.amount
+            parent = tax_model.search([('children_tax_ids', 'in', [tax.id])])
+            if parent:
+                main_tax = parent
+                aliquota = parent.amount
+            else:
+                main_tax = tax
+            kind_id = main_tax.kind_id.id
+            payability = main_tax.payability
+            imposta = tax_line.amount
+            base = tax_line.base
+            if main_tax.id not in tax_grouped:
+                tax_grouped[main_tax.id] = {
+                    'ImponibileImporto': 0,
+                    'Imposta': imposta,
+                    'Aliquota': aliquota,
+                    'Natura_id': kind_id,
+                    'EsigibilitaIVA': payability,
+                    'Detraibile': 0.0,
+                }
+            else:
+                tax_grouped[main_tax.id]['Imposta'] += imposta
+            if tax.account_id:
```

Comment:
Nel caso di imposte indetraibili totalmente o parzialmente, il conto configurato nell'imposta è vuoto ma va utilizzato il conto della imposte raggruppate (campo `tax_line.account_id`).
```suggestion
            if tax.account_id or tax_line.account_id:
```
cc @stevech091 correggimi se ho detto qualche strafalcione, grazie

Reply:
Il conto configurato nell'imposta madre è vuoto: è valorizzato in quella figlia (e `tax` in questo caso sarebbe la figlia).

Puoi descrivere i passi per riprodurre il problema?

Reply:
@eLBati vedi https://github.com/OCA/l10n-italy/pull/1005#issuecomment-487559656

Reply:
@stevech091 @SimoRubi ho creato questa fattura

![image](https://user-images.githubusercontent.com/1033131/56897098-df40a700-6a8d-11e9-86df-8cc59ef09677.png)

e nella comunicazione fatture ottengo

![image](https://user-images.githubusercontent.com/1033131/56897135-f67f9480-6a8d-11e9-89aa-f820c239ed59.png)

che mi pare corretto

Reply:
ok grazie, abbiamo scoperto che il cliente aveva l'imposta configurata senza la riga (D) quindi sbagliata